### ISSUE TITLE
Ideal value of Global Scale Var : Struct2Depth

### ISSUE THREAD

**Issue by ezorfa** [tensorflow/models#8061]
Please go to Stack Overflow for help and support:

http://stackoverflow.com/questions/tagged/tensorflow

Also, please understand that many of the models included in this repository are experimental and research-style code. If you open a GitHub issue, here is our policy:

1. It must be a bug, a feature request, or a significant problem with documentation (for small docs fixes please send a PR instead).
2. The form below must be filled out.

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **What is the top-level directory of the model you are using**: Struct2Depth
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: yes in pytorch
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: Not needed
- **TensorFlow version (use command below)**: Not needed
- **Bazel version (if compiling from source)**: Not needed
- **CUDA/cuDNN version**: 10
- **GPU model and memory**: GeForce 11GB
- **Exact command to reproduce**: None

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

`python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`

### Describe the problem
Hi @aneliaangelova !

Could you please let me know what should be the ideal value of global scale var, when its trained?

I am trying to implement "object constraint loss" in pytorch but facing difficulties. The scale var I defined slowly decreases to zero. I posted a similar question on stackoverflow : https://stackoverflow.com/questions/59800247/pytorch-equivalent-of-tf-variable

Thankyou!

**Comment by aneliaangelova** [comment]
The scale variable (if you are referring to the object size) is actually several such variables, one for each class, e.g. pedestrians, cars, etc; each class is training a separate variable.

**Comment by ezorfa** [comment]
@aneliaangelova Could you please refer me in the code where this is happening. I can only see at one place a single "global_scale_var" is defined. or Am I missing something?

**Comment by ezorfa** [comment]
@aneliaangelova  

Relating to my previous comment, here are the links that show that self.global_scale_var is defined only once:

https://github.com/tensorflow/models/blob/6a3bcef8ee40aaf359cb7162f057b9543ce05822/research/struct2depth/model.py#L344

https://github.com/tensorflow/models/blob/6a3bcef8ee40aaf359cb7162f057b9543ce05822/research/struct2depth/model.py#L140

**Comment by aneliaangelova** [comment]
Sorry I thought you are referring to the variables for object size which are also learnable. 
I am not sure why you're having trouble with a pytorch implementation. Were you able to run the code we provided in TF?

**Comment by ezorfa** [comment]
@aneliaangelova I actually dint try TF impl. Nevertheless, Could you please refer me to the place in the code  where you said " the variables for object size which are also learnable" because I see no such thing happening.

**Comment by VincentCa** [comment]
Hi @ezorfa,
We have multi-class models that were implemented with different object size variables for each class. However, for the open-source version of the code we only use a single "global_scale_var", which is why you find it defined as a scalar rather than a vector here. If I remember correctly, we found performance differences to be negligible, because the object size constraint is most useful to resolve long-range scenarios (esp. forward motion of objects moving at very similar speeds as the observer). Since this pretty much only applies to cars or similarly motorized vehicles, this seemed sufficient to us.

Unfortunately, I don't have a readily available record anymore of what value it typically converged to, but we did relate it to world units and if you do train with multi-class, they should be consistent to each other in relative terms (i.e. trucks > pedestrians > cars with reasonable ratios). It shouldn't converge to zero indeed, but it might be hard to catch the problem in pytorch. Do your depth inference results look reasonable during training? Sometimes the net might get stuck in degenerate states, it might, for example, predict all-zero depth if your learning rate is too high.
-Vincent
